.github/workflows/run_g_tests.yml

```diff
@@ -6,15 +6,39 @@ on:
 
 jobs:
   run_g_tests:
-    name: Run G Tests
-    runs-on: self-hosted
+    name: Run G Tests LV${{ matrix.version }}
+    runs-on: 
+      - self-hosted
+      - windows
+      - x64
+      - rdss-measlinkbot-win-10-lv64
+    strategy:
+      matrix:
+        version: ["2020", "2021", "2022", "2023"]
+      fail-fast: false
+    timeout-minutes: 90
     steps:
+    - name: Enable git long paths
+      run: git config  --global core.longpaths true
+
     - uses: actions/checkout@v4
 
     - uses: actions/setup-python@v5
       with:
         python-version: '3.10'
 
+    - name: Generate port number
+      id: port_number
+      run: |
+        if (${{ matrix.version }} -eq "2020") {
```

Comment:
FYI, you could specify version/port combinations using strategy/matrix/include, but I think the if/else approach is also ok.
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations